internal/parse.go

```diff
@@ -0,0 +1,15 @@
+package internal
+
+import "regexp"
+
+// MatchCaptureGroups takes a regular expression and string and returns all of the named capture group results in a map.
+func MatchCaptureGroups(regEx *regexp.Regexp, str string) map[string]string {
+	match := regEx.FindStringSubmatch(str)
+	results := make(map[string]string)
+	for i, name := range regEx.SubexpNames() {
+		if i > 0 && i <= len(match) {
```

Comment:
This is **_super_** small 😄  — for my own clarity — can `i <= len(match)` just be `i < len(match)`, if `i` never gets incremented up to the number of items in the slice?

Reply:
And actually — is the part of the condition after `&&` even necessary? `range` handles the incrementing of `i` for this loop and the exiting of the loop when all slice items have been returned

Reply:
This is a good catch.

`i>0` is not necessary, correct. 

The second term is needed; note that this is a range over `regEx.SubexpNames()` which result is mostly independent than that returned from `regEx.FindStringSubmatch(str)`... the only dependence is that the order of matches corresponds to the order of the names, and it is important to only return the **first** instance of each (so we ignore multiple instances of the capture group names).

@Toure since this is migrating existing code I'd vote you probably don't need to fix this in this PR. I can follow up with a PR to add tests for this function and clean up a bit.